oracle/config/api.go

```diff
@@ -17,6 +17,10 @@ type APIConfig struct {
 	// Interval is the interval at which the provider should update the prices.
 	Interval time.Duration `json:"interval"`
 
+	// ReconnectTimeout is the amount of time the provider should wait before
+	// reconnecting to the API.
+	ReconnectTimeout time.Duration `json:"reconnectTimeout"`
```

Comment:
This might be particularly useful if we want to stop requesting if we get rate limited.